Rewrite article "View a JSON file or server response with formatting" (#3368)
MicrosoftDocs/edge-developer: microsoft-edge/web-platform/json-viewer.md

---

> title: View a JSON file as-is or formatted

Reviewer: I don't know that the "as-is" part of the title is needed. The value of the tool _is_ that it can format JSON nicely. The fact that a JSON file or response appears in a browser tab as-is doesn't require any tool. That's the default way that browsers tend to show text files.

So I suggest changing the title to "View a formatted JSON file", or similar.

Author: alts:
* View a formatted JSON file
* View a JSON file, with formatting

Author: result, global:
```
# View a JSON file, with formatting
```

> description: View a formatted or minified JSON file as-is or formatted, in Microsoft Edge.

Reviewer: ```suggestion
description: View a potentially minified JSON file formatted, in Microsoft Edge.
```

Author: result:
```
description: How to view a minified JSON file or JSON response with formatting 
for readability, or a formatted JSON file, in Microsoft Edge.
```

author: MSEdgeTeam
ms.author: msedgedevrel
ms.topic: conceptual
ms.service: microsoft-edge
ms.subservice: devtools
ms.date: 03/05/2025
---
# View a JSON file as-is or formatted

> When you open a JSON file in Microsoft Edge, the file is initially displayed as-is (nested, or minified), and the **Pretty-print** checkbox is initially cleared.

Reviewer: ```suggestion
When you open a JSON HTTP response or a JSON file in Microsoft Edge, the content is initially displayed as-is. For example, if the JSON was minified, with its entire content on a single line and empty spaces removed, the JSON appears minified when displayed in Microsoft Edge initially.

When Microsoft Edge detects that the HTTP response or file displayed in a browser tab is JSON data, a **Pretty-print** checkbox is displayed. The **Pretty-print** checkbox is initially cleared.
```

Author: result:
```
When you open a JSON HTTP response or a JSON file in Microsoft Edge, the content is initially displayed as-is.  For example, if the JSON was minified, with its entire content on a single line and empty spaces removed, the JSON appears minified when initially displayed in Microsoft Edge.

To display a minified JSON file in a more readable format, select the **Pretty-print** checkbox.
```

A `.json` file that contains whitespace and nesting ([128KB.json](https://microsoftedge.github.io/Demos/json-dummy-data/128KB.json)), displayed as-is:

> ![A .json file that's non-minified, initially opened in Edge](./json-viewer-images/non-minified-json-file-as-is.png)

Reviewer: I think we need to remove these 2 lines. If the JSON file is already nicely formatted, there's no need for the tool. The user will already be able to easily view the content.

Author: removed png & 
```
A `.json` file that contains whitespace and nesting ([128KB.json](https://microsoftedge.github.io/Demos/json-dummy-data/128KB.json)), displayed as-is:

![A .json file that's non-minified, initially opened in Edge](./json-viewer-images/non-minified-json-file-as-is.png)
```

A `.json` file that's minified ([128KB-min.json](https://microsoftedge.github.io/Demos/json-dummy-data/128KB-min.json)), displayed as-is:

![A .json file that's minified, initially opened in Edge](./json-viewer-images/minified-json-file-as-is.png)

<!-- ====================================================================== -->
## The Pretty-print checkbox

> Select or clear the **Pretty-print** checkbox to change between viewing the `.json` file using a nested, easy-to-read display or a concatenated, dense display.

Reviewer: ```suggestion
Select or clear the **Pretty-print** checkbox to change between viewing the JSON content (a file or HTTP response) using a easy-to-read, multiline, format or a concatenated, dense display.
```

Author: streamlining this a lot

A `.json` file that contains whitespace and nesting ([128KB.json](https://microsoftedge.github.io/Demos/json-dummy-data/128KB.json)), displayed with the **Pretty-print** checkbox selected and then cleared:

![A .json file that's non-minified, viewed as minified](./json-viewer-images/non-minified-json-file-non-pretty-printed.png)

A `.json` file that's minified ([128KB-min.json](https://microsoftedge.github.io/Demos/json-dummy-data/128KB-min.json)), displayed with the **Pretty-print** checkbox selected:

![A .json file that's minified, viewed with Pretty-print selected](./json-viewer-images/minified-json-file-pretty-printed.png)

<!-- ------------------------------ -->
<!-- #### Video: The JSON viewer -->

<!-- [![Thumbnail image for video "The JSON viewer"](./json-viewer-images/json-viewer.png)](https://www.youtube.com/watch?v=DO7V6lw1SWA) -->

<!-- ====================================================================== -->
## Supported scenarios

> The **JSON viewer** is not supported in windows that are opened by using the `window.open` JavaScript method. In windows opened with `window.open`, JSON data is displayed as a single line of text, without formatting or syntax highlighting.

Reviewer: Remove. This doesn't seem to be an issue anymore.

<!-- ====================================================================== -->
## View reformatted JSON server responses

To view a JSON response from a web server as reformatted JSON:

1. Open a new tab or window in Microsoft Edge.

1. Type a URL that returns JSON data in the address bar. For example, use this sample JSON response: [https://microsoftedge.github.io/Demos/json-dummy-data/256KB-min.json](https://microsoftedge.github.io/Demos/json-dummy-data/256KB-min.json).

   You can find more samples of JSON responses at [MicrosoftEdge / Demos > json-dummy-data](https://microsoftedge.github.io/Demos/json-dummy-data/).

> 1. Microsoft Edge detects that the returned data is JSON and formats it automatically.

Reviewer: ```suggestion
1. Microsoft Edge detects that the returned data is JSON and displays the **Pretty-print** checkbox.

1. Click **Pretty-print** to display the JSON content in an easy-to-read format.
```

<!-- ====================================================================== -->
## View reformatted JSON files

To view a JSON file stored on your device as reformatted JSON:

1. Open a new tab or window in Microsoft Edge.

1. Press **Ctrl+O** on Windows and Linux, or **Command+O** on macOS, and then select a JSON file.

> 1. Microsoft Edge detects that the file contains JSON data and formats it automatically.

Reviewer: Same change as my previous comment. This doesn't happen automatically.

Also, I wonder if we should merge these two h2 sections. The whole tool is just one checkbox, it feels like we can get away with a shorter article, and just have a section that says that the tool works for both files opened from disk _and_ JSON responses from servers.

Author: near top:
This feature works for files that are opened from disk and for JSON responses from servers.
